function/registry/registry.go

```diff
@@ -333,3 +337,62 @@ func NewOperator(op string, operator func(float64, float64) float64) function.Me
 		},
 	}
 }
+
+var graphiteSelect = function.MetricFunction{
+	Name:         "graphite",
+	MinArguments: 1,
+	MaxArguments: 1,
+	Compute: func(context function.EvaluationContext, args []function.Expression, groups function.Groups) (function.Value, error) {
+		graphiteQuery, err := args[0].Evaluate(context)
+		if err != nil {
+			return nil, err
+		}
+		graphitePattern, err := graphiteQuery.ToString()
+		if err != nil {
+			return nil, err
+		}
+		graphiteAPI, ok := context.API.(api.GraphiteStore)
+		if !ok {
+			return nil, fmt.Errorf("your API does not support graphite queries")
+		}
+		results := graphite.GetGraphiteMetrics(graphitePattern, graphiteAPI)
+		if len(results) == 0 {
+			return nil, fmt.Errorf("graphite query %s resulted in no graphite metrics", graphiteQuery)
+		}
+		ok = context.FetchLimit.Consume(len(results))
```

Comment:
we should apply the predicates here, before we consume, and before we fetch. it's wasteful to fetch everything if we're going to throw them away later.